tests/test_transformations/test_arcsinh_transformation.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+import numpy as np
+import pandas as pd
+import anndata
+from spac.transformations import arcsinh_transformation
+
+
+class TestArcsinhTransformation(unittest.TestCase):
+
+    def setUp(self):
+        data_df = pd.DataFrame({
+            'Feature1': [2, 4, 6],
+            'Feature2': [5, 10, 15]
+        })
+        var_df = pd.DataFrame(index=data_df.columns)
+        self.adata = anndata.AnnData(X=data_df.values, var=var_df)
+
+        # Modify data for layer1 to make it different from main matrix
+        layer1_data = data_df.values + np.array([[1, -1], [-1, 2], [0, 0]])
+        self.adata.layers["layer1"] = layer1_data
+
+    def test_arcsinh_transformation_main_matrix(self):
+        transformed_adata = arcsinh_transformation(self.adata)
+        # Hardcoded expected transformation using 20th percentile as co-factor
+        expected_data = np.array([
+            [0.6643306, 0.6643306],
+            [1.15447739, 1.15447739],
+            [1.50575679, 1.50575679]
+        ])
+        np.testing.assert_array_almost_equal(
+            transformed_adata.layers['arcsinh'], expected_data
+        )
+
+    def test_arcsinh_transformation_layer(self):
+        transformed_adata = arcsinh_transformation(
+            self.adata, input_layer="layer1")
+        # Hardcoded expected transformation using 20th percentile
+        # as co-factor on layer1
+        expected_data = np.array([
+            [0.88137359, 0.530343],
+            [0.88137359, 1.283796],
+            [1.44363548, 1.480294]
+        ])
+        np.testing.assert_array_almost_equal(
+            transformed_adata.layers['arcsinh'], expected_data
+        )
+
+    def test_arcsinh_transformation_fixed_co_factor(self):
```

Comment:
@fangliu117 , would you please add a test of cofactor taking precedence over quantile?